swap/src/protocol/bob/event_loop.rs

```diff
@@ -250,11 +250,22 @@ impl EventLoopHandle {
     }
 
     pub async fn request_spot_price(&mut self, btc: bitcoin::Amount) -> Result<monero::Amount> {
-        Ok(self
+        let response = self
             .spot_price
             .send_receive(spot_price::Request { btc })
-            .await?
-            .xmr)
+            .await?;
+
+        match (response.xmr, response.error) {
+            (Some(xmr), None) => Ok(xmr),
+            (_, Some(error)) => {
+                bail!(error);
+            }
+            (None, None) => {
+                bail!(
+                    "Unexpected response for spot-price request, neither price nor error received"
+                );
```

Comment:
Does it not bother you that you have to write this code? :D

If you model your response differently, this can be avoided entirely.

Reply:
😅 it was late at night